hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AbstractFSWAL.java

```diff
@@ -872,6 +874,33 @@ private IOException convertInterruptedExceptionToIOException(final InterruptedEx
     }
   }
 
+  public void archive(RegionServerServices services) throws IOException{
+    if (getNumRolledLogFiles() < 1) {
+      return;
+    }
+    // get the earliest log of this WAL instance
+    Map.Entry<Path, WalProps> firstWALEntry = this.walFile2Props.firstEntry();
+    // flush reigons if necessary
+    Map<byte[], List<byte[]>> regions =
+      this.sequenceIdAccounting.findLower(firstWALEntry.getValue().encodedName2HighestSequenceId);
+    if (regions != null) {
+      for (Map.Entry<byte[], List<byte[]>> entry : regions.entrySet()) {
+        String encodedRegionName = Bytes.toString(entry.getKey());
+        HRegion r = (HRegion) services.getRegion(encodedRegionName);
+        if (r == null) {
+          LOG.warn("Failed to flush of {} when archive manually, because it is not online on us",
+            encodedRegionName);
+          return;
+        }
+        r.flushcache(entry.getValue(), false, FlushLifeCycleTracker.DUMMY);
+      }
+    }
+    // move the log file to archive dir
+    this.totalLogSize.addAndGet(-firstWALEntry.getValue().logSize);
+    moveLogFileToArchiveDir(firstWALEntry.getKey());
```

Comment:
What if IOException is thrown from this call ?
Would totalLogSize be correct ?

Reply:
Good point, the move action should be called first. 
BTW, this part copied from existing code, maybe need a follow-on jira to fix it too.
Thanks.